tests/integration/golang/aim/run/search_test.go

```diff
@@ -0,0 +1,523 @@
+	})
+	assert.Nil(s.T(), err)
+	_, err = s.tagFixtures.CreateTag(context.Background(), &models.Tag{
+		Key:   "aim.runName",
+		Value: "TestRunTag1",
+		RunID: run1.ID,
+	})
+	assert.Nil(s.T(), err)
+	_, err = s.metricFixtures.CreateLatestMetric(context.Background(), &models.LatestMetric{
+		Key:       "TestMetric",
+		Value:     1.1,
+		Timestamp: 1234567890,
+		Step:      1,
+		IsNan:     false,
+		RunID:     run1.ID,
+		LastIter:  1,
+	})
+	assert.Nil(s.T(), err)
+	_, err = s.paramFixtures.CreateParam(context.Background(), &models.Param{
+		Key:   "param1",
+		Value: "value1",
+		RunID: run1.ID,
+	})
+	assert.Nil(s.T(), err)
+
+	run2, err := s.runFixtures.CreateRun(context.Background(), &models.Run{
+		ID:         "id2",
+		Name:       "TestRun2",
+		UserID:     "2",
+		Status:     models.StatusScheduled,
+		SourceType: "JOB",
+		StartTime: sql.NullInt64{
+			Int64: 111111111,
+			Valid: true,
+		},
+		EndTime: sql.NullInt64{
+			Int64: 222222222,
+			Valid: true,
+		},
+		ExperimentID:   *experiment.ID,
+		ArtifactURI:    "artifact_uri2",
+		LifecycleStage: models.LifecycleStageDeleted,
+	})
+	assert.Nil(s.T(), err)
+	_, err = s.tagFixtures.CreateTag(context.Background(), &models.Tag{
+		Key:   "aim.runName",
+		Value: "TestRunTag2",
+		RunID: run2.ID,
+	})
+	assert.Nil(s.T(), err)
+	_, err = s.metricFixtures.CreateLatestMetric(context.Background(), &models.LatestMetric{
+		Key:       "TestMetric",
+		Value:     2.1,
+		Timestamp: 1234567890,
+		Step:      1,
+		IsNan:     false,
+		RunID:     run2.ID,
+		LastIter:  1,
+	})
+	assert.Nil(s.T(), err)
+	_, err = s.paramFixtures.CreateParam(context.Background(), &models.Param{
+		Key:   "param2",
+		Value: "value2",
+		RunID: run2.ID,
+	})
+	assert.Nil(s.T(), err)
+
+	run3, err := s.runFixtures.CreateRun(context.Background(), &models.Run{
+		ID:         "id3",
+		Name:       "TestRun3",
+		UserID:     "3",
+		Status:     models.StatusRunning,
+		SourceType: "JOB",
+		StartTime: sql.NullInt64{
+			Int64: 333444444,
+			Valid: true,
+		},
+		EndTime: sql.NullInt64{
+			Int64: 444555555,
+			Valid: true,
+		},
+		ExperimentID:   *experiment.ID,
+		ArtifactURI:    "artifact_uri3",
+		LifecycleStage: models.LifecycleStageActive,
+	})
+	assert.Nil(s.T(), err)
+	_, err = s.tagFixtures.CreateTag(context.Background(), &models.Tag{
+		Key:   "aim.runName",
+		Value: "TestRunTag3",
+		RunID: run3.ID,
+	})
+	assert.Nil(s.T(), err)
+	_, err = s.metricFixtures.CreateLatestMetric(context.Background(), &models.LatestMetric{
+		Key:       "TestMetric",
+		Value:     3.1,
+		Timestamp: 1234567890,
+		Step:      1,
+		IsNan:     false,
+		RunID:     run3.ID,
+		LastIter:  1,
+	})
+	assert.Nil(s.T(), err)
+	_, err = s.paramFixtures.CreateParam(context.Background(), &models.Param{
+		Key:   "param3",
+		Value: "value3",
+		RunID: run3.ID,
+	})
+	assert.Nil(s.T(), err)
+	tests := []struct {
+		name    string
+		request request.SearchRunsRequest
+		runs    []*models.Run
+	}{
```

Comment:
Can you please add tests for this functions as well:
- `match` function.
- `search` function.

Reply:
also I know, it could be a pain, but it will be really nice to cover all supported fields in our `query.go`

Reply:
also, it will be nice to have like a one complex query, which can use several fields and operators